src/main/cc/interop.cc

```diff
@@ -325,6 +325,35 @@ namespace skija {
         }
     }
 
+    namespace PaintFilterCanvas {
+        JavaVM* _vm;
+        jmethodID onFilterId;
+
+        void onLoad(JavaVM* vm, JNIEnv* env) {
+            _vm = vm;
+            jclass local = env->FindClass("org/jetbrains/skija/PaintFilterCanvas");
+            onFilterId = env->GetMethodID(local, "onFilter", "(J)Z");
+        }
+
+        void onUnload(JNIEnv* env) {
+        }
+
+        bool onFilter(jobject obj, SkPaint& paint) {
+            JNIEnv *env;
+            _vm->AttachCurrentThread((void **) &env, NULL);
+            jboolean result = env->CallBooleanMethod(obj, onFilterId, &paint);
+            _vm->DetachCurrentThread();
+            return result;
+        }
+
+        void dispose(jobject obj) {
+            JNIEnv *env;
+            _vm->AttachCurrentThread((void **) &env, NULL);
+            env->DeleteGlobalRef(obj);
```

Comment:
Who creates this global reference?

Reply:
Was in PointFilterCanvas.cpp, attachToJava()

Moved here for symmetry.